subprojects/core/src/main/java/org/gradle/internal/scan/config/BuildScanPluginAutoApply.java

```diff
@@ -0,0 +1,107 @@
+ * - No request is added if the project has already requested the build scan plugin, via `plugins` or `buildscript`.
+ * - The plugin request is inserted before any other plugin requests.
+ * - A fixed version of the build scan plugin is requested.
+ */
+public class BuildScanPluginAutoApply implements PluginRequestsTransformer {
+    public static final String BUILD_SCAN_PLUGIN_AUTO_APPLY_VERSION = "1.10";
+    private static final PluginId BUILD_SCAN_PLUGIN_ID = DefaultPluginId.of("com.gradle.build-scan");
+    private static final String BUILD_SCAN_PLUGIN_GROUP = "com.gradle";
+    private static final String BUILD_SCAN_PLUGIN_NAME = "build-scan-plugin";
+
+    private final StartParameter startParameter;
+
+    public BuildScanPluginAutoApply(StartParameter startParameter) {
+        this.startParameter = startParameter;
+    }
+
+    @Override
+    public PluginRequests transformPluginRequests(PluginRequests requests, Object pluginTarget) {
+        if (!startParameter.isBuildScan()) {
+            return requests;
+        }
+
+        if (!isRootProject(pluginTarget)) {
+            return requests;
+        }
+
+        if (isPluginAlreadyRequested(requests, (Project) pluginTarget)) {
+            return requests;
+        }
+
+        DefaultPluginRequest buildScanPluginRequest = new DefaultPluginRequest(BUILD_SCAN_PLUGIN_ID, BUILD_SCAN_PLUGIN_AUTO_APPLY_VERSION, true, 0, "auto-apply", null);
+        return prependRequest(buildScanPluginRequest, requests);
+    }
+
+    private PluginRequests prependRequest(PluginRequestInternal buildScanPluginRequest, PluginRequests requests) {
+        List<PluginRequestInternal> copyRequests = Lists.newArrayList();
+        copyRequests.add(buildScanPluginRequest);
+        CollectionUtils.addAll(copyRequests, requests);
+        return new DefaultPluginRequests(copyRequests);
+    }
+
+    private boolean isRootProject(Object pluginTarget) {
+        if (!(pluginTarget instanceof Project)) {
+            return false;
+        }
+
+        Project project = (Project) pluginTarget;
+        return project.getGradle().getParent() == null
+            && project == project.getRootProject();
+    }
+
+    private boolean isPluginAlreadyRequested(PluginRequests requests, Project project) {
+        for (PluginRequestInternal request : requests) {
+            if (BUILD_SCAN_PLUGIN_ID.equals(request.getId())) {
+                // Build scan plugin already requested in `plugins`
+                return true;
+            }
+        }
+
+        Configuration classpathConfiguration = project.getBuildscript().getConfigurations().getByName(ScriptHandler.CLASSPATH_CONFIGURATION);
+        for (Dependency dependency : classpathConfiguration.getDependencies()) {
+            // Build scan plugin already included as `classpath` dependency
+            if (BUILD_SCAN_PLUGIN_GROUP.equals(dependency.getGroup()) && BUILD_SCAN_PLUGIN_NAME.equals(dependency.getName())) {
```

Comment:
:o: Potentially should handle the case where an earlier build scan plugin version is used. In those situations the version of this class should override the one in the build script. Alternatively, a warning message should be rendered.